include/boost_1_73_0/patches/boost/container/string.hpp

```diff
@@ -0,0 +1,3505 @@
+//////////////////////////////////////////////////////////////////////////////
```

Comment:
Is this some kind of a  patched version of `boost::container::string`?

Reply:
Yes, as the 1.0.73 boost::string errors out because of an implicit cost cast.

Reply:
@dutow Although the fix you did is 100% legit (the `const` should definitely be there), adding this custom version of `boost/container/string.hpp` in my opinion will be really hard to maintain properly. If Oracle decides to update boost to a newer version we will definitely forget about this :(
Moreover, it seems to me that the original issue you were trying to fix was not an error, but a warning
```
*(unsigned char*)&hdr = *(const unsigned char*)&this->members_.m_repr;
```
Even the latest boost 1.78.0
https://github.com/boostorg/container/blob/boost-1.78.0/include/boost/container/string.hpp#L263
does not have this `const`  - I doubt they could miss this with their test coverage.
So, what I suggest, is to remove this patched version of `string.hpp` and suppress this warning with `ADD_CXX_COMPILE_FLAGS_TO_FILES(...)` in the `CMakeLists.txt`